src/cunumeric/index/zip_omp.cc

```diff
@@ -46,7 +47,9 @@ struct ZipImplBody<VariantKind::OMP, DIM, N> {
         for (size_t idx = 0; idx < volume; ++idx) {
           Legion::Point<N> new_point;
           for (size_t i = 0; i < N; i++) {
-            new_point[i] = compute_idx(indx_ptrs[i][idx], shape[i]);
+            auto pair        = compute_idx_omp(indx_ptrs[i][idx], shape[i]);
+            new_point[i]     = pair.first;
+            is_out_of_bounds = is_out_of_bounds || pair.second;
```

Comment:
Can we switch this snippet to something like:
```
if (pair.second) is_out_of_bounds = true;
```
This way we only actually do the atomic write if the flag is true, which should be a very rare occurrence.

Similar for other cases where we set an "out-of-bounds" flag in an OpenMP task, including for wrap.

Reply:
makes sense. pushing the change